cmd/kube-proxy/app/server.go

```diff
@@ -257,8 +304,31 @@ func (o *Options) Run() error {
 	if err != nil {
 		return err
 	}
+	o.proxyServer = proxyServer
+	return o.runLoop()
+}
+
+// runLoop will watch on the update change of the proxy server's configuration file.
+// Return an error when updated
+func (o *Options) runLoop() error {
+	if o.watcher != nil {
+		o.watcher.Run()
+	}
+
+	// run the proxy in goroutine
+	go func() {
+		err := o.proxyServer.Run()
+		o.errCh <- err
+	}()
 
-	return proxyServer.Run()
+	for {
+		select {
+		case err := <-o.errCh:
+			if err != nil {
```

Comment:
I _think_ this is the cause for https://github.com/kubernetes/kubernetes/issues/76727 since `kube-proxy --cleanup` returns nil error and is expected to exit. cc @vllry @thockin

Reply:
Yep. #76732 is an open change for that, will comment there.